test/e2e/serial/tests/tolerations.go

```diff
@@ -350,6 +351,83 @@ var _ = Describe("[serial][disruptive][slow][rtetols] numaresources RTE tolerati
+					if err != nil {
+						return err
+					}
+
+					updatedNode, updated, err := taints.AddOrUpdateTaint(node, tnt)
+					if err != nil {
+						return err
+					}
+					if !updated {
+						return nil
+					}
+
+					klog.Infof("adding taint: %q to node: %q", tnt.String(), updatedNode.Name)
+					err = fxt.Client.Update(ctx, updatedNode)
+					if err != nil {
+						return err
+					}
+					return nil
+				}).WithPolling(1 * time.Second).WithTimeout(1 * time.Minute).ShouldNot(HaveOccurred())
+				targetNodeNames = append(targetNodeNames, taintedNode.Name)
+				klog.Infof("considering node: %q tainted with %q", taintedNode.Name, tnt.String())
+
+				By("trigger an RTE pod restart on the tainted node by deleting the pod")
+				ds := appsv1.DaemonSet{}
+				err = fxt.Client.Get(ctx, client.ObjectKey(dsKey), &ds)
+				Expect(err).ToNot(HaveOccurred())
+
+				klog.Info("verify RTE pods before triggering the restart still include the pod on the tainted node")
+				pods, err := podlist.With(fxt.Client).ByDaemonset(ctx, ds)
+				Expect(err).NotTo(HaveOccurred(), "Unable to get pods from daemonset %q: %v", ds.Name, err)
+				Expect(len(pods)).To(Equal(len(workers)), "pods number is not as expected for RTE daemonset: expected %d found %d", len(workers), ds.Status.DesiredNumberScheduled)
+
+				var podToDelete corev1.Pod
+				for _, pod := range pods {
+					if pod.Spec.NodeName == taintedNode.Name {
+						podToDelete = pod
+						break
+					}
+				}
+				Expect(podToDelete).NotTo(BeNil())
+
+				klog.Infof("delete the pod %s/%s of the tainted node", podToDelete.Namespace, podToDelete.Name)
+				err = fxt.Client.Delete(ctx, &podToDelete)
+				Expect(err).ToNot(HaveOccurred())
+				err = wait.With(fxt.Client).Timeout(2*time.Minute).ForPodDeleted(ctx, podToDelete.Namespace, podToDelete.Name)
+				Expect(err).ToNot(HaveOccurred(), "pod %s/%s still exists", podToDelete.Namespace, podToDelete.Name)
+
+				klog.Info(fmt.Sprintf("waiting for daemonset %v to report correct pods' number", dsKey.String()))
+				updatedDs, err := wait.With(fxt.Client).Interval(time.Second).Timeout(time.Minute).ForDaemonsetPodsCreation(ctx, dsKey, len(workers)-1)
+				Expect(err).NotTo(HaveOccurred(), "pods number is not as expected for RTE daemonset: expected %d found %d", len(workers)-1, updatedDs)
+				updatedDs, err = wait.With(e2eclient.Client).Interval(10*time.Second).Timeout(3*time.Minute).ForDaemonSetReadyByKey(ctx, dsKey)
+				Expect(err).ToNot(HaveOccurred(), "failed to get the daemonset ready: %v", err)
+
+				klog.Info("verify there is no RTE pod running on the tainted node")
+				pods, err = podlist.With(fxt.Client).ByDaemonset(ctx, *updatedDs)
+				Expect(err).NotTo(HaveOccurred(), "Unable to get pods from Daemonset %q: %v", ds.Name, err)
+				for _, pod := range pods {
+					Expect(pod.Spec.NodeName).NotTo(Equal(taintedNode.Name), "found RTE pod running on tainted node without toleration on NROP obj")
+				}
+			})
+
```

Comment:
This works. Maybe we can do better: get all the pods running on the tainted node, verify none of them is a RTE pod (there's a label for that, and we can also check the command line and/or the image)

Reply:
that is also another way to do this check. could you please tell why getting all node's pods and check the label on each pod is better than checking the assigned node of the all the RTE pods? 
one thing I see better in the current way is that it will need to go over less pods (3 vs 30+) considering our CI supports at best 3 workers.

Reply:
the main reason is I'd like to encourage variety in the test flow. I'd like to crosscheck results coming from different angles, so the final check could (and IMO should) take a different route than keep depending more and more on daemonset reports.

Checking all pods on a selected node is not something concerning performance wise, if the test code or even worse our platform can't cope with the load caused by listing all pods on a node we have way worse issues in our hands.

Reply:
okay I see your point now. That is indeed good idea. I'll improve this part as you suggested. Thanks for calrifying.